base/server/systemd/SystemdStartupNotifier.java

```diff
@@ -0,0 +1,59 @@
+// --- BEGIN COPYRIGHT BLOCK ---
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; version 2 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// (C) 2020 Red Hat, Inc.
+// All rights reserved.
+// --- END COPYRIGHT BLOCK ---
+
+package com.netscape.cmscore.systemd;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+
+import com.netscape.cmscore.apps.StartupNotifier;
+
+public class SystemdStartupNotifier implements StartupNotifier {
+    interface Systemd extends Library {
+        public int sd_booted();
+        public int sd_notify(int unset_environment, String state);
+    }
+
+    /* loadLibrary is deprecated in jna >= 5; replaced with load()
+     * which avoids the cast.  But RHEL 8 has jna 4.5 so we must put
+     * up with deprecation warnings on Fedora. */
+    static Systemd systemd = (Systemd) Native.loadLibrary("systemd", Systemd.class);
+
+    static boolean isNotifyService = hasNotifySocket() && systemdBooted();
+
+    private static boolean systemdBooted() {
+        return systemd.sd_booted() > 0;
+    }
+
+    private static boolean hasNotifySocket() {
+        return System.getenv("NOTIFY_SOCKET") != null;
+    }
+
+    private static void notify(String status) {
+        if (isNotifyService) {
+            System.out.println("Notifying systemd:\n" + status);
+            systemd.sd_notify(0 /* unset_environment */, status);
+        } else {
+            System.err.println("Failed to notify systemd (isNotifyService = false)");
```

Comment:
In ordinary configuration they don't - but I can include the subsystem name in this message to disambiguate (if we stay with this approach rather than switching to Tomcat `LifecycleListener`).

Reply:
I'm modifying the `StartupNotifier` interface to return the result to the caller.  So `CMSEngine` will process the result and write log messages to its own logger.  `StartupNotifier` no longer writes to stderr or stdout.